### PR TITLE
Disallow passing config to TypeAdapter for models and implement config for TypedDict

### PR DESCRIPTION
**Pull request by adriangb**
Requires pydantic-core update.

cc @samuelcolvin 

Selected Reviewer: @Kludex

### CODE REVIEW

File: docs/usage/errors.md [hunk 1]
```diff
@@ -773,4 +773,48 @@ try:
 except PydanticUserError as exc_info:
     assert exc_info.code == 'invalid_annotated_type'
 ```
+
+## `config` is unused with TypeAdapter {#type-adapter-config-unused}
+
+You will get this error if you try to pass `config` to `TypeAdapter` when the type is a type that has it's own config that cannot be overridden (currently this is only `BaseModel`, `TypedDict` and dataclasses):
```
Comment (samuelcolvin) [inline docs/usage/errors.md:779]:
```suggestion
You will get this error if you try to pass `config` to `TypeAdapter` when the type is a type that has it's own config that cannot be overridden (currently this is only `BaseModel`, `TypedDict` and `dataclass`):
```

File: docs/usage/errors.md [hunk 2]
```diff
@@ -773,4 +773,48 @@ try:
...
+
+```py
+from typing_extensions import TypedDict
+
+from pydantic import ConfigDict, PydanticUserError, TypeAdapter
+
+
+class Model(TypedDict):
+    pass
```
Comment (samuelcolvin) [inline docs/usage/errors.md:788]: probably easier to add a field.

File: docs/usage/errors.md [hunk 3]
```diff
@@ -773,4 +773,48 @@ try:
...
+
+You will get this error if you try to pass `config` to `TypeAdapter` when the type is a type that has it's own config that cannot be overridden (currently this is only `BaseModel`, `TypedDict` and dataclasses):
+
+```py
+from typing_extensions import TypedDict
+
+from pydantic import ConfigDict, PydanticUserError, TypeAdapter
+
+
+class Model(TypedDict):
```
Comment (samuelcolvin) [inline docs/usage/errors.md:787]: less confusing/missable to a user if you call this `MyTypedDict`.

File: pydantic/_internal/_generate_schema.py [hunk 4]
```diff
@@ -20,6 +20,8 @@
 from pydantic_core import CoreSchema, core_schema
 from typing_extensions import Annotated, Final, Literal, TypedDict, get_args, get_origin, is_typeddict
 
+from pydantic.config import ConfigDict
```
Comment (samuelcolvin) [inline pydantic/_internal/_generate_schema.py:23]:
```suggestion
from ..config import ConfigDict
```

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM. Please update.

**Comment by adriangb** [comment]
please review

**Comment by samuelcolvin** [comment]
merging to fix in #5942

**Pull request merged**